lib/spack/spack/package_base.py

```diff
@@ -201,9 +201,12 @@ def __init__(cls, name, bases, attr_dict):
         if hasattr(cls, "executables") or hasattr(cls, "libraries"):
             # Append a tag to each detectable package, so that finding them is faster
             if not hasattr(cls, "tags"):
-                setattr(cls, "tags", [DetectablePackageMeta.TAG])
+                cls.tags = [DetectablePackageMeta.TAG]
             elif DetectablePackageMeta.TAG not in cls.tags:
-                cls.tags.append(DetectablePackageMeta.TAG)
+                if "tags" in attr_dict:
+                    cls.tags.append(DetectablePackageMeta.TAG)
+                else:
+                    cls.tags = cls.tags[:] + [DetectablePackageMeta.TAG]
```

Comment:
This is the fix, the other parts are packages added in the PR from which this was extracted that revealed the bug + needed modification to tests